core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -1007,6 +1007,19 @@ public boolean canAllow(CarbonTable carbonTable, TableOperation operation) {
     return indexColumn;
   }
 
+  /**
+   * Whether this table supports flat folder structure, it means all data files directly written
+   * under table path
+   */
+  public boolean isSupportFlatFolder() {
+    boolean supportFlatFolder = Boolean.parseBoolean(CarbonCommonConstants.DEFAULT_FLAT_FOLDER);
+    Map<String, String> tblProps = getTableInfo().getFactTable().getTableProperties();
+    if (tblProps.containsKey(CarbonCommonConstants.FLAT_FOLDER)) {
```

Comment:
Can the user set FLAT_FOLDER property in CarbonProperties(applicable for all tables) instead of specifying for each table??

Reply:
Because of old tables also the property will be applicable if property set at System level, that causes hybrid storage like some data in segment folders and some data under table path. thats why it is not set at system level